plugin-energy-estimation-tdp/src/lib.rs

```diff
@@ -48,48 +48,37 @@ impl AlumetPlugin for EnergyEstimationTdpPlugin {
 
     fn init(config: ConfigTable) -> anyhow::Result<Box<Self>> {
         let config = deserialize_config(config)?;
-        Ok(Box::new(EnergyEstimationTdpPlugin {
-            config: Some(config),
-            metrics: Arc::new(Mutex::new(Metrics::default())),
-        }))
+        Ok(Box::new(EnergyEstimationTdpPlugin { config }))
     }
 
     fn start(&mut self, alumet: &mut alumet::plugin::AlumetPluginStart) -> anyhow::Result<()> {
-        let mut metrics = self.metrics.lock().unwrap();
-        let config: Config = self.config.take().unwrap();
-
         // Create the energy attribution metric and add its id to the
         // transform plugin metrics' list.
-        metrics.pod_estimate_attributed_energy = Some(alumet.create_metric(
+        let pod_estimate_attributed_energy_metric = alumet.create_metric(
             "pod_estimate_attributed_energy",
             Unit::Joule,
-            "Energy consumption estimated to the pod",
-        )?);
+            "Pod's estimated energy consumption",
+        )?;
 
-        // Add the transform now but fill its metrics later.
-        alumet.add_transform(Box::new(EnergyEstimationTdpTransform::new(
-            config,
-            self.metrics.clone(),
-        )));
-        Ok(())
-    }
+        let cpu_usage = self.config.as_ref().unwrap().cpu_usage_per_pod.clone();
+        let config_cpy = self.config.take().unwrap();
 
-    fn pre_pipeline_start(&mut self, alumet: &mut AlumetPreStart) -> anyhow::Result<()> {
-        /// Finds the RawMetricId with the name of the metric.
-        /// Will only run once, just before the pipeline starts.
-        fn find_metric_by_name(alumet: &mut AlumetPreStart, name: &str) -> anyhow::Result<RawMetricId> {
-            let (id, _metric) = alumet
-                .metrics()
-                .into_iter()
-                .find(|m| m.1.name == name)
-                .expect(&format!("Cannot find metric {name}, is the 'rapl' plugin loaded?").to_string());
-            Ok(id.to_owned())
-        }
-
-        // Lock the metrics mutex to apply its modifications.
-        let mut metrics = self.metrics.lock().unwrap();
-
-        metrics.cpu_usage_per_pod = Some(find_metric_by_name(alumet, "cgroup_cpu_usage_total")?);
+        // Add the transform now but fill its metrics later.
+        alumet.add_transform_builder(move |ctx| {
+            let name = ctx.transform_name("energy_estimation_tdp");
+
+            let cpu_usage_metric = ctx
+                .metric_by_name(&cpu_usage)
+                .with_context(|| format!("Metric not found : {}", cpu_usage))?
```

Comment:
nitpick: error contexts are usually lowercase (by convention)
```suggestion
                .with_context(|| format!("metric not found : {}", cpu_usage))?
```

Reply:
done